rust/migrate-wicked/src/interface.rs

```diff
@@ -546,34 +541,71 @@ impl Interface {
             }
             if let Some(routes) = &ipv6_static.routes {
                 for route in routes {
-                    if let Some(nexthops) = &route.nexthops {
-                        // TODO fix when implementing better route handling
-                        // the logged warning isn't really true for multiple hops
-                        // as gateways just can't have multiple nexthops AFAICT
-                        if gateway6.is_some() || nexthops.len() > 1 {
-                            connection_result.warnings.push(anyhow::anyhow!(
-                                "Multipath routing isn't natively supported by NetworkManager"
-                            ));
-                        } else {
-                            gateway6 = Some(IpAddr::from_str(&nexthops[0].gateway).unwrap());
+                    new_routes6.push(match route.try_into() {
+                        Ok(route) => route,
+                        Err(e) => {
+                            connection_result.warnings.push(e);
+                            continue;
                         }
-                    }
+                    });
                 }
             }
         }
 
+        let routes4 = if !new_routes4.is_empty() {
+            Some(new_routes4)
+        } else {
+            None
+        };
+        let routes6 = if !new_routes6.is_empty() {
+            Some(new_routes6)
+        } else {
+            None
+        };
+
         connection_result.ip_config = IpConfig {
             addresses,
             method4,
             method6,
-            gateway4,
-            gateway6,
+            routes4,
+            routes6,
             ..Default::default()
         };
         Ok(connection_result)
     }
 }
 
+impl TryFrom<&Route> for IpRoute {
+    type Error = anyhow::Error;
+    fn try_from(route: &Route) -> Result<Self, Self::Error> {
+        let mut next_hop: Option<IpAddr> = None;
+        if let Some(nexthops) = &route.nexthops {
+            if nexthops.len() > 1 {
+                return Err(anyhow::anyhow!(
+                    "Multipath routing isn't natively supported by NetworkManager"
+                ));
+            } else {
+                next_hop = Some(IpAddr::from_str(&nexthops[0].gateway).unwrap());
+            }
+        }
+        let destination = if route.destination.is_some() {
+            IpInet::from_str(route.destination.clone().unwrap().as_str())?
+        } else if next_hop.is_some() {
+            // In this case the route is a default route, so we need to give the destination a /0 as prefix length
+            // Note the ip address doesn't actually really matter to NetworkManager
+            IpInet::new(next_hop.unwrap(), 0)?
```

Comment:
If it's not matter, why not writing it like this, as it more readable/understandable?
```suggestion
            IpInet::new("0.0.0.0", 0)?
```

Reply:
Well my reasoning was that it gives more information where the route "came from". But yea thinking about it now I agree, the nexthop should be enough and `0.0.0.0` is indeed more clear that it is a gateway.